tests/test_main.py

```diff
@@ -30,3 +36,131 @@ def test_set_base64_default_line_size_rejects_negative_values(self):
         with self.assertRaises(ValueError):
             xmlsec.base64_default_line_size(-1)
         self.assertEqual(xmlsec.base64_default_line_size(), size)
+
+
+class TestCallbacks(base.TestMemoryLeaks):
+    def setUp(self):
+        super().setUp()
+        xmlsec.cleanup_callbacks()
+
+    @given(funcs=strategies.lists(
```

Comment:
I suppose the reported memleaks are caused by the `hypothesis` usage - I recall playing with it some time ago and seeing objects not being cleaned up, presumably some nested lambdas in `hypothesis` internals. Maybe just use the `pytest.mark.parametrize` for now since the strategy draws fixed examples anyway?

Reply:
I've removed hypothesis from tests, but I'm still getting `AssertionError: unreferenced objects detected`. It appear only to happen on the tests for signing failures, but I can't really work out much more than that right now. I've had another look at my new code, and I can't see anything that would leave a reference count incremented. It happens when looking at the test failures even if no Python callbacks have been registered.

Any help would be much appreciated.